Assets/Mirror/Components/NetworkLobbyManager.cs

```diff
@@ -0,0 +1,498 @@
+using System.Collections.Generic;
+using System.Linq;
+using UnityEngine;
+using UnityEngine.SceneManagement;
+using UnityEngine.Serialization;
+
+namespace Mirror
+{
+    [AddComponentMenu("Network/NetworkLobbyManager")]
+    [HelpURL("https://vis2k.github.io/Mirror/Components/NetworkLobbyManager")]
+    public class NetworkLobbyManager : NetworkManager
+    {
+        struct PendingPlayer
+        {
+            public NetworkConnection conn;
+            public GameObject lobbyPlayer;
+        }
+
+        // configuration
+        [Header("Lobby Settings")]
+        [FormerlySerializedAs("m_ShowLobbyGUI")] [SerializeField] internal bool showLobbyGUI = true;
+        [FormerlySerializedAs("m_MinPlayers")] [SerializeField] int minPlayers = 1;
+        [FormerlySerializedAs("m_LobbyPlayerPrefab")] [SerializeField] NetworkLobbyPlayer lobbyPlayerPrefab;
+
+        [Scene]
+        public string LobbyScene;
+
+        [Scene]
+        public string GameplayScene;
+
+        // runtime data
+        [FormerlySerializedAs("m_PendingPlayers")] List<PendingPlayer> pendingPlayers = new List<PendingPlayer>();
+        List<NetworkLobbyPlayer> lobbySlots = new List<NetworkLobbyPlayer>();
+
+        public bool allPlayersReady = false;
+
+        public override void OnValidate()
+        {
+            maxConnections = Mathf.Max(maxConnections, 0); // always >= 0
+
+            minPlayers = Mathf.Min(minPlayers, maxConnections); // always <= maxConnections
+            minPlayers = Mathf.Max(minPlayers, 0); // always >= 0
+
+            if (lobbyPlayerPrefab != null)
+            {
+                NetworkIdentity identity = lobbyPlayerPrefab.GetComponent<NetworkIdentity>();
+                if (identity == null)
+                {
+                    lobbyPlayerPrefab = null;
+                    Debug.LogWarning("LobbyPlayer prefab must have a NetworkIdentity component.");
```

Comment:
Should this be an error instead of warning?

Reply:
Changed